main/php_network.h

```diff
@@ -162,7 +162,7 @@ PHPAPI int php_poll2(php_pollfd *ufds, unsigned int nfds, int timeout);
 /* timeval-to-timeout (for poll(2)) */
 static inline int php_tvtoto(struct timeval *timeouttv)
 {
-	if (timeouttv) {
+	if (timeouttv && timeouttv->tv_sec >= 0 && timeouttv->tv_sec <= (INT_MAX / 1000)) {
 		return (timeouttv->tv_sec * 1000) + (timeouttv->tv_usec / 1000);
```

Comment:
In lack of something like `TIME_T_MAX`, `INT_MAX` seems to be a prudent choice. However, the following addition can still overflow, even if we assume that `tv_usec` is less than 1,000,000 (not sure if that is guaranteed by any standard, though). So maybe checking that `tv_sec` is less than or equal `((INT_MAX - 1000) / 1000)` is safer. And after all, a timeout that large (a couple of weeks) is highly unlikely to be useful in practice.